service/history/workflow/task_generator.go

```diff
@@ -248,9 +248,14 @@ func (r *TaskGeneratorImpl) GenerateDirtySubStateMachineTasks(
 	stateMachineRegistry *hsm.Registry,
 ) error {
 	tree := r.mutableState.HSM()
+	// Early return here to avoid accessing the transition history. It may be disabled via dynamic config.
+	outputs := tree.Outputs()
+	if len(outputs) == 0 {
+		return nil
+	}
 	transitionHistory := r.mutableState.GetExecutionInfo().TransitionHistory
 	versionedTransition := transitionHistory[len(transitionHistory)-1]
```

Comment:
may be better to add some sanity check here to ensure transition history is enabled in case the dc got messed up in prod.

Reply:
oh the code will panic and it will be captured and fail the state transition. we should be good.